### PR TITLE
OCaml Changelog

### PR DESCRIPTION
**Pull request by tmattio**
This PR introduces the [OCaml Changelog](https://staging.ocaml.org/changelog) - a feed of the newest releases and feature highlights of the OCaml Compiler, OCaml Platform and other official projects including Ppxlib and Omd.

The goal of the Changelog is to offer a page where OCaml developers can stay up to date with the latest official releases, and most importantly quickly learn about the most important new features available and how to use them.

At this stage, the OCaml Changelog is an experiment. We'd like to merge as such to facilitate communication with everyone involved, especially the maintainers of the projects featured in the Changelog.

In particular, if we decide on supporting the OCaml Changelog in the long term, we'll need to find a good release process with the maintainers of the tools that are featured in the Changelog, to make sure that every release gets an announcement and that important features are highlighted.

We have talked about this with several maintainers, but at this point, we have not agreed on a definitive process. While the Changelog remains in an experimental stage, for demonstration purposes, we intend to take on the workload of creating the changelog entries and working with the maintainers to write the announcements, until we've built a consensus on the final process.

Note that the introduction of the Changelog impacts the content that is available on the OCaml Blog.

The previous News section, which contained the newsletters (Multicore and Compiler), but also releases and articles from the OCaml Platform Blog, now only contains newsletters. The Platform Blog articles are now part of the OCaml Planet feed (the aggregate of community blog posts), and the releases are part of the OCaml Changelog.

**TODO**

- [x] Make the experimental nature of the page clear in the UI
- [x] Filter by tags
- [ ] Load more to avoid loading all of the changelog entries
- [x] Add a link from the homepage

### CODE REVIEW

**Comment by avsm** [comment]
I like this a lot! Especially when the tags are available, this is a brilliant way to keep up with platform and compiler activity.  I think we should also be able to create a discuss.ocaml.org reserved category into which such announcements could be automatically mirrored (for discussion purposes). 

Regarding a mechanism for creating the updates, how about using the opam-repository metadata as the primary way to keep track of the update contents?  We could define an `x-changes` field that (like description) contains the body of the changes involved in that release, which could then be slurped into ocaml.org. 

(adding x-ref to https://github.com/ocaml/infrastructure/issues/20, and you might also want to respond with a pointer this issue on https://discuss.ocaml.org/t/users-of-opam-repo-what-sort-of-change-tracking-would-you-like/11590 to see if we can get more discussion going there from other interested users)

**Comment by tmattio** [comment]
Thanks a lot for the feedback @avsm!

This is ready to review/merge. I've added a note on the page to make it clear that this is experimental and that we'll work with the projects maintainers on the release process.

In the meantime, let's create an issue to track the different feedback we received:

- Better distinction of releases vs features (e.g. changing the icon has been mentioned)
- Add a load more button to not load the whole page at once
- Link to individual changelogs
- Create a category on Discuss that is synced with the changelog to have discussions on the announcements
- Consider featuring user-facing OCaml infrastructure changes in the changelog
- Consider pulling changelogs from the opam metadata (e.g. with an x-change field that's populated by publication tools like dune-release)

I'll wait for your approval @sabine, but I'll merge the PR myself so I can post the announcement on Discuss simultaneously.

File: .github/workflows/ci.yml
Comment (sabine) [inline .github/workflows/ci.yml:38]: make sure to prominently mention when merging that the opam-repository pin is updated (in the commit message body)

File: CONTRIBUTING.md
Comment (sabine) [inline CONTRIBUTING.md]: Omd has been deprecated in favor of Ppxlib? :thinking: That's what I read here. I am confused both on whether that's true and also what this means.

File: src/ocamlorg_frontend/pages/blog.eml
Comment (sabine) [inline src/ocamlorg_frontend/pages/blog.eml]: On first glance, I disagree with this. We should not be rendering the Planet blog posts on ocaml.org. Linking out to the blogs is more respectful to the community, and the posts will likely also be more readable in the context of their original page.

I recommend we don't make this change together with the Changelog addition. Pulling in the community blog posts should be a separate PR - if this is even something that we and the community want. Reason: this is a feature unrelated to the changelog and it will be helpful to have a separate diff for that patch.

Also, the URLs are broken: I see http://127.0.0.1:8080/blog/https://tarides.com/blog/2023-03-02-the-journey-to-ocaml-multicore-bringing-big-ideas-to-life

**Review by sabine — CHANGES_REQUESTED** [review]
Overall the addition of the changelog looks good to me. No complaints on that.

However, pulling in and rendering on ocaml.org the OCaml Planet blog posts should be a separate PR.

File: CONTRIBUTING.md
Comment (tmattio) [reply CONTRIBUTING.md]: Sorry, should be [OMP](https://github.com/ocaml-ppx/ocaml-migrate-parsetree), not Omd.

File: src/ocamlorg_frontend/pages/blog.eml
Comment (tmattio) [reply src/ocamlorg_frontend/pages/blog.eml]: To clarify, this is not rendering community blog posts on OCaml.org, we still link to the websites if there's a link, in particular for all the posts we fetched from the RSS feed. We only render the blog posts that are hosted on OCaml.org, for instance, the post on signing opam-repository.

Comment (sabine) [reply src/ocamlorg_frontend/pages/blog.eml]: Ah ok, then I was misreading that. Only URL needs to be fixed then

File: CONTRIBUTING.md
Comment (sabine) [reply CONTRIBUTING.md]: Ok good, that makes a lot more sense

File: src/ocamlorg_frontend/pages/blog.eml
Comment (tmattio) [reply src/ocamlorg_frontend/pages/blog.eml]: And IMO this should be part of this PR: moving the opam and OCaml releases out of `news/` to the Changelog makes the current news section confusing, with essentially blog posts with the Multicore newsletter.

I can make two PRs if it's easier to review but from a user perspective making the news section of the blog clearer should be part of the changelog introduction I think.

File: CONTRIBUTING.md
Comment (tmattio) [reply CONTRIBUTING.md]: Fixed, thanks for spotting it.

File: src/ocamlorg_frontend/pages/blog.eml
Comment (tmattio) [reply src/ocamlorg_frontend/pages/blog.eml]: Fixed

Comment (sabine) [reply src/ocamlorg_frontend/pages/blog.eml]: Yes, it's working now and good to go.

**Review by sabine — APPROVED** [review]

**Pull request merged**
